src/pages/api/modules/usenet/queue.ts

```diff
@@ -33,42 +35,116 @@ async function Get(req: NextApiRequest, res: NextApiResponse) {
+        });
+
+        if (!nzbgetStatus) {
+          throw new Error('Error while getting NZBGet status');
+        }
+
+        const nzbgetItems: UsenetQueueItem[] = nzbgetQueue.map((item: NzbgetQueueItem) => ({
+          id: item.NZBID.toString(),
+          name: item.NZBName,
+          progress: (item.DownloadedSizeMB / item.FileSizeMB) * 100,
+          eta: (item.RemainingSizeMB * 1000000) / nzbgetStatus.DownloadRate,
+          // Multiple MB to get bytes
+          size: item.FileSizeMB * 1000 * 1000,
+          state: getNzbgetState(item.Status),
+        }));
+
+        response = {
+          items: nzbgetItems,
+          total: nzbgetItems.length,
+        };
+        break;
+      }
+      case 'Sabnzbd': {
+        if (!service.apiKey) {
+          throw new Error(`API Key for service "${service.name}" is missing`);
+        }
 
-    const { origin } = new URL(service.url);
-    const queue = await new Client(origin, service.apiKey).queue(offset, limit);
-
-    const items: UsenetQueueItem[] = queue.slots.map((slot) => {
-      const [hours, minutes, seconds] = slot.timeleft.split(':');
-      const eta = dayjs.duration({
-        hour: parseInt(hours, 10),
-        minutes: parseInt(minutes, 10),
-        seconds: parseInt(seconds, 10),
-      } as any);
-
-      return {
-        id: slot.nzo_id,
-        eta: eta.asSeconds(),
-        name: slot.filename,
-        progress: parseFloat(slot.percentage),
-        size: parseFloat(slot.mb) * 1000 * 1000,
-        state: slot.status.toLowerCase() as any,
-      };
-    });
-
-    const response: UsenetQueueResponse = {
-      items,
-      total: queue.noofslots,
-    };
+        const { origin } = new URL(service.url);
+        const queue = await new Client(origin, service.apiKey).queue(offset, limit);
+
+        const items: UsenetQueueItem[] = queue.slots.map((slot) => {
+          const [hours, minutes, seconds] = slot.timeleft.split(':');
+          const eta = dayjs.duration({
+            hour: parseInt(hours, 10),
+            minutes: parseInt(minutes, 10),
+            seconds: parseInt(seconds, 10),
+          } as any);
+
+          return {
+            id: slot.nzo_id,
+            eta: eta.asSeconds(),
+            name: slot.filename,
+            progress: parseFloat(slot.percentage),
+            size: parseFloat(slot.mb) * 1000 * 1000,
+            state: slot.status.toLowerCase() as any,
+          };
+        });
+
+        response = {
+          items,
+          total: queue.noofslots,
+        };
+        break;
+      }
+      default:
+        throw new Error(`Service type "${service.type}" unrecognized.`);
+    }
 
     return res.status(200).json(response);
   } catch (err) {
     return res.status(500).send((err as any).message);
   }
 }
 
+function getNzbgetState(status: string) {
+  switch (status) {
+    case 'QUEUED':
+      return 'queued';
+    case 'PAUSED ':
+        return 'paused';
+    default:
+      return 'downloading';
+  }
+}
+
```

Comment:
Some translation should be added here

Reply:
This is just a helper function to map the NZBget status to the `UsenetQueueItem` status options. From what I can tell, it isn't every displayed on screen.

Reply:
oh so it's just an internal conversion, my bad